### PR TITLE
Fix Yahoo provider & set static rate for base currency #16446 

### PR DESCRIPTION
**Pull request by martinvy**
https://is.roihunter.com/issues/16446

@business-factory/pythonistas please review & merge.

### CODE REVIEW

File: gold_digger/data_providers/yahoo.py [hunk 1]
```diff
@@ -38,55 +38,67 @@ def get_supported_currencies(self, date_of_exchange):
...
+                'Time': '8:59am',
+                'id': 'GBP=X'
+            }
         """
         if response:
             try:
-                results = response.json()["query"]["results"]
-                return results["rate"] if results else []
+                rates = response.json()["query"]["results"]["rate"]
+                return rates
```
Comment (harpener) [inline gold_digger/data_providers/yahoo.py]: @martinvy You could have written this without the variable :-)

File: gold_digger/managers/exchange_rate_manager.py [hunk 2]
```diff
@@ -132,6 +139,18 @@ def get_exchange_rate_by_date(self, date_of_exchange, from_currency, to_currency
...
 
+    def _get_sum_of_rates_in_period(self, start_date, end_date, currency):
+        """
+        :type start_date: datetime.date
+        :type end_date: datetime.date
+        :type currency: str
+        :rtype: list[(str,int,Decimal)]
+        """
+        if currency == self._base_currency:
+            return [("BASE", 1, ExchangeRate.base(self._base_currency).rate),]
```
Comment (harpener) [inline gold_digger/managers/exchange_rate_manager.py]: @martinvy Could you specify the `BASE` as some constant? Like `BASE_PROVIDER`, `FAKE_PROVIDER`, `DEFAULT_PROVIDER` or something? :-) It looks like it is here for some reason but anything can be here really.

File: gold_digger/managers/exchange_rate_manager.py [hunk 3]
```diff
@@ -148,26 +167,32 @@ def get_average_exchange_rate_by_dates(self, start_date, end_date, from_currency
...
-            if to_count != number_of_days:
-                self._logger.warning("Provider %s miss %s days with currency %s while range request on %s - %s" %
-                                     (to_provider, number_of_days - to_count, to_currency, start_date, end_date))
+            self._logger.info(
+                "Sum of currencies %s (%s records) = %s, %s (%s records) = %s in period %s - %s by (%s, %s)",
+                from_currency, from_count, from_sum, to_currency, to_count, to_sum, start_date, end_date, from_provider, to_provider
+            )
+            if from_count != number_of_days and from_currency != self._base_currency:
+                self._logger.warning(
+                    "Provider %s misses %s days with currency %s while range request on %s - %s",
```
Comment (harpener) [inline gold_digger/managers/exchange_rate_manager.py]: @martinvy Just so you know, phrase `is missing` feels more natural to me, but it seems that this is allowed as well :-D

File: gold_digger/managers/exchange_rate_manager.py [hunk 4]
```diff
@@ -148,26 +167,32 @@ def get_average_exchange_rate_by_dates(self, start_date, end_date, from_currency
...
             if from_count and from_sum and to_count and to_sum:
                 from_average = from_sum / from_count
                 to_average = to_sum / to_count
                 conversion = 1 / from_average
                 return Decimal(to_average * conversion)
+
             self._logger.error("Date range 'count' and/or 'sum' are empty")
 
-        self._logger.debug("Range request failed: from %s to %s" % (_from_currency, _to_currency))
+        self._logger.exception("Range request failed: from %s to %s", _from_currency, _to_currency)
```
Comment (harpener) [inline gold_digger/managers/exchange_rate_manager.py]: @martinvy You are using exception level when there is no exception context, You should use error here :-)

File: tests/data_providers/test_yahoo.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        }
+        """
+    yahoo._get = lambda *a, **kw: sample
+
+    rates = yahoo.get_all_by_date(date.today(), {"EUR", "CZK", "EEK"})
+
+    assert rates == {
+        "EUR": Decimal("0.8555"),
+        "CZK": Decimal("22.0606"),
+    }
```
Comment (harpener) [inline tests/data_providers/test_yahoo.py]: @martinvy Can you please add one blank line here? :-)

File: gold_digger/managers/exchange_rate_manager.py (hunk 2 above)
Comment (martinvy) [reply gold_digger/managers/exchange_rate_manager.py]: @harpener It's there just for logging purposes, nothing more.

**Pull request merged**